app/controllers/order/international_order.js

```diff
@@ -0,0 +1,141 @@
+import Ember from "ember";
+import SearchOptionMixin from "stock/mixins/search_option";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import { regex } from "stock/constants/regex";
+
+export default Ember.Controller.extend(SearchOptionMixin, AsyncMixin, {
+  i18n: Ember.inject.service(),
+  showError: false,
+  orderService: Ember.inject.service(),
+
+  shipmentorCarryoutCode: Ember.computed({
+    get() {
+      return this.get("shipmentorCarryoutCode");
+    },
+    set(_, value) {
+      return value;
+    }
+  }),
+
+  isInvalidDescription: Ember.computed("orderDescription", function() {
+    return !this.get("orderDescription");
+  }),
+
+  isInvalidCode: Ember.computed("shipmentorCarryoutCode", function() {
+    const codeRegEx = new RegExp(regex.SHIPMENT_ORDER_REGEX);
+    return (
+      !this.get("shipmentorCarryoutCode") ||
+      !codeRegEx.test(this.get("shipmentorCarryoutCode"))
+    );
+  }),
+
+  isInValidCountry: Ember.computed("country", function() {
+    return !this.get("country");
+  }),
+
+  isInvalidShipmentDate: Ember.computed("shipmentDate", function() {
+    return !this.get("shipmentDate");
+  }),
+
+  getCurrentUser: Ember.computed(function() {
+    var store = this.get("store");
+    var currentUser = store.peekAll("user_profile").get("firstObject") || null;
+    return currentUser;
+  }).volatile(),
+
+  shipmentTypes: Ember.computed(function() {
+    let translation = this.get("i18n");
+    let shipment = translation.t("order_transports.shipment");
+    let carryout = translation.t("order_transports.carry_out");
+
+    return [
+      { name: shipment, id: "Shipment" },
+      { name: carryout, id: "CarryOut" }
+    ];
+  }),
+
+  formatOrderCode(type, code) {
+    return type == "CarryOut" ? `C${code}` : `S${code}`;
+  },
+
+  orderParams() {
+    let validCode = this.formatOrderCode(
+      this.get("selectedType.id"),
+      this.get("shipmentorCarryoutCode")
+    );
+
+    let params = {
+      state_event: "submit",
```

Comment:
Its better to have strings like "submit", "Shipment", "CarryOut" etc to move into a constant file.
For example: if "submit" needs to be changed to "Submit", then it will be present in only one file, rather than changing in multiple file